packages/parse5/lib/tokenizer/index.ts

```diff
@@ -925,7 +953,7 @@ export class Tokenizer {
     //------------------------------------------------------------------
     private _stateData(cp: number): void {
         this.preprocessor.dropParsedChunk();
-        this.ctLoc = this._getCurrentLocation();
+        this.updateStartLocation();
```

Comment:
feels a bit weird having to do this book-keeping — its existence seems to indicate that the data might be out of date and before you access it you’d have to call this function?

Reply:
I played around with this some more. Turns out we can actually get rid of location updates in the *DATA states altogether: #402